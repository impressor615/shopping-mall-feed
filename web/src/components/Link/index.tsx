import "./index.scss";

import classnames from "classnames";
import React from "react";
import { Link } from "react-router-dom";

interface ButtonProps {
  children: React.ReactNode;
  to: string;
  active?: boolean;
  [rest: string]: any;
}

const MyLink = ({ children, to, active, ...rest }: ButtonProps) => (
  <Link
    to={to}
    className={classnames("link", { active })}
    {...rest}
  >
    { children }
  </Link>
);

export default MyLink;
