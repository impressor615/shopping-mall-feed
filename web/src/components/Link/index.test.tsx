import enzyme from "enzyme";
import React from "react";

import Link from "@/components/Link";

describe("Link", () => {
  const children = "link";
  it("should render children", () => {
    const wrapper = enzyme.shallow(<Link to="/products">{ children }</Link>);
    expect(wrapper.find(".link").children().text()).toEqual(children);
  });

  it("should render active link when active prop is true", () => {
    const wrapper = enzyme.shallow(<Link active to="/products">{ children }</Link>);
    expect(wrapper.find(".link.active")).toHaveLength(1);
  });

  it("should be matched with snapshot", () => {
    const wrapper = enzyme.shallow(<Link to="/products">{ children }</Link>);
    expect(wrapper).toMatchSnapshot();
  });
});
