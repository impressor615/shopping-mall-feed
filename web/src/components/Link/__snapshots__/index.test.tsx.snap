// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should be matched with snapshot 1`] = `
<Link
  className="link"
  replace={false}
  to="/products"
>
  link
</Link>
`;
